src/DynamoApplications/PathResolvers.cs

```diff
@@ -58,13 +60,41 @@ public IEnumerable<string> PreloadedLibraryPaths
 
         public string UserDataRootFolder
         {
-            get { return string.Empty; }
+            get { return Path.Combine(Environment.GetFolderPath(
+                Environment.SpecialFolder.ApplicationData),
+                "Dynamo", "Dynamo Core").ToString(); }
         }
 
         public string CommonDataRootFolder
         {
             get { return string.Empty; }
         }
+
+        /// <summary>
+        /// Returns the full path of user data location of all version of this
+        /// Dynamo product installed on this system. The default implementation
+        /// returns list of all subfolders in %appdata%\Dynamo as well as 
+        /// %appdata%\Dynamo\Dynamo Core\ folders.
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<string> GetDynamoUserDataLocations()
```

Comment:
Because

-  all integrators always pass `StartConfiguration` which includes PathResolver
- PathResolver already contains some folder information

I think it makes better sense to implement `GetDynamoUserDataLocations()` as part of `PathResolver`